Packs/jamf/Integrations/jamfV2/jamfV2.py

```diff
@@ -74,12 +83,26 @@ def _generate_token(self) -> str:
         """
         Generates new token.
         """
-        resp = self._http_request(method='POST', url_suffix='api/v1/auth/token', resp_type='json')
-        token = resp.get('token')
-        expiration_time = int(dateparser.parse(resp.get('expires')).timestamp())  # type: ignore[union-attr]
+        if self.basic_auth_flag:  # basic auth is used to create a token
+            resp = self._http_request(method='POST', url_suffix='api/v1/auth/token', resp_type='json',
+                                      auth=(self.username, self.password))
+            token = resp.get('token')
+            expiration_time = int(dateparser.parse(resp.get('expires')).timestamp())
+        else:  # client id and client secret are used to create a token
```

Comment:
I would rather be explicit on this else statement so if we get a new auth method in the future it wont require changing that as well. If we can take this logic to different functions its even better.